lib/consumer/presentation/consumer_home_page.dart

```diff
@@ -1,29 +1,56 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:connectivity/connectivity.dart';
+import 'package:internet_connection_checker/internet_connection_checker.dart';
 
 import 'package:deall/core/presentation/widgets/drawer_widget.dart';
 import 'package:deall/consumer/shared/providers.dart';
 import 'package:deall/consumer/presentation/retailer_listview.dart';
 import 'package:deall/consumer/presentation/widgets/search_bar_filter.dart';
 
-
 class ConsumerHomePage extends ConsumerStatefulWidget {
   const ConsumerHomePage({Key? key}) : super(key: key);
 
   @override
-  ConsumerState<ConsumerStatefulWidget> createState() => _ConsumerHomePageState();
-
-
+  ConsumerState<ConsumerStatefulWidget> createState() =>
+      _ConsumerHomePageState();
 }
 
 class _ConsumerHomePageState extends ConsumerState<ConsumerHomePage> {
+  StreamSubscription<ConnectivityResult>? subscription;
+
   @override
   void initState() {
     super.initState();
-    Future.microtask(() => ref.read(retailerListNotifierProvider.notifier).getRetailerList());
+    Future.microtask(() =>
+        ref.read(retailerListNotifierProvider.notifier).getRetailerList());
   }
+
+  @override
+  void dispose() {
+    super.dispose();
+    subscription!.cancel();
+  }
+
+  void connectivityChecker() {
+    subscription = Connectivity()
+        .onConnectivityChanged
+        .listen((ConnectivityResult result) async {
+      if (result == ConnectivityResult.none) {
+          ScaffoldMessenger.of(context).showSnackBar(
+            const SnackBar(
+              content: Text("No connection"),
+            ),
+          );
+      }
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
+    connectivityChecker();
     final mq = MediaQuery.of(context);
     return Scaffold(
       appBar: enterLocationAppBar(),
```

Comment:
I would suggest u put your listener in initstate.

case 1: The user's device has no connection when logged in
1. The listener will not perform any action when the device is not connected.
2. The notifier check and return a newly created RetailerListState.noConnection() instead of failure
3. In the presentation, when the state is noConnection, display a text in the scaffold telling the user that they are not connected

4. when the user connects to the internet, the getRetailerList will get triggered in the listener and the listener will be disposed right after the method is called.

What u think?

Reply:
okay i'll try that